framework/base/Controller.php

```diff
@@ -221,6 +221,13 @@ public function createAction($id)
         }
 
         $actionMap = $this->actions();
+        $this->ensureBehaviors();
+        foreach ($this->getBehaviors() as $behavior) {
+            if ($behavior instanceof ControllerBehaviorInterface) {
+                $actionMap = array_replace_recursive($actionMap, $behavior->actions());
```

Comment:
Could change it to use `ArrayHelper::merge()` and add a new helper `yii\helpers\ArrayValueIfKeyNotExists` that would only add the key/value pair if the key doesn't exist in the other array.

```php
$array1 = [
        'error' => 'yii\web\ErrorAction',
        'view' =>  'yii\web\ViewAction',
 ];
$array2 = [
        'error' => 'yii\web\ErrorAction',
        'view' => new \yii\helpers\ArrayValueIfKeyNotExists(  'my\custom\ViewAction'),
        'custom' => new \yii\helpers\ArrayValueIfKeyNotExists(  'my\custom\CustomAction'),
 ];

$result = ArrayHelper::merge($array1, $array2);

// result will be...
[
        'error' => 'yii\web\ErrorAction',
        'view' =>  'yii\web\ViewAction',
        'custom' => 'my\custom\CustomAction',
 ];
```

This would provide the most flexibility as it would allow the developer to decide whether or not the values should be replaced.